pdl-compiler/src/analyzer.rs

```diff
@@ -190,6 +190,12 @@ pub enum ErrorCode {
     E42 = 42,
     E43 = 43,
     DuplicateDefaultTag = 44,
+    InvalidOptionalField = 45,
+    UndeclaredConditionIdentifier = 46,
+    InvalidConditionIdentifier = 47,
+    InvalidConditionValue = 48,
+    E49 = 49,
```

Comment:
Just a small question: I've not yet understood why we have `Enn` enum variants? Could you ask an AI chatbot to come up with a good name if the problem is a lack of imagination? :smile:

Reply:
Aha... I had not understood this so far :smile: I think you need an `Other` enum variant for all those errors then. If they're not important enough to document or to even name, they cannot be very important for the overall API?

Reply:
I like having unique IDs for errors :)

1. it is easier to test since I can check that a code snippet fails with a specific error
2. the error number is printed as part of the diagnostic, users could in theory refer to the `test_eN` test case to understand what it's about

Reply:
I see... then just leave it like that until others come and ask :slightly_smiling_face: